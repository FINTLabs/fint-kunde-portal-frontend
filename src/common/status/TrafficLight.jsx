import React, { Component } from "react";
import { styled } from '@mui/material/styles';
import { Tooltip, CircularProgress } from "@mui/material";
const PREFIX = 'TrafficLight';

const classes = {
  failed: `${PREFIX}-failed`,
  ok: `${PREFIX}-ok`,
  partiallyFailed: `${PREFIX}-partiallyFailed`,
  progress: `${PREFIX}-progress`
};

const Root = styled('div')((
  {
    theme
  }
) => ({
  [`& .${classes.failed}`]: {
    height: "25px",
    width: "25px",
    backgroundColor: theme.palette.primary.main,
    borderRadius: "50%",
    display: "inline-block",
    webkitBoxShadow:
      "0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
    mozBoxShadow:
      "0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
    boxShadow:
      "0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)"
  },

  [`& .${classes.ok}`]: {
    height: "25px",
    width: "25px",
    backgroundColor: theme.palette.secondary.main,
    borderRadius: "50%",
    display: "inline-block",
    webkitBoxShadow:
      "0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
    mozBoxShadow:
      "0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
    boxShadow:
      "0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)"
  },

  [`& .${classes.partiallyFailed}`]: {
    height: "25px",
    width: "25px",
    backgroundColor: "#ff9800",
    borderRadius: "50%",
    display: "inline-block",
    webkitBoxShadow:
      "0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
    mozBoxShadow:
      "0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
    boxShadow:
      "0px 3px 5px -1px rgba(0, 0, 0, 0.2), 0px 6px 10px 0px rgba(0, 0, 0, 0.14), 0px 1px 18px 0px rgba(0, 0, 0, 0.12)"
  },

  [`& .${classes.progress}`]: {}
}));

class TrafficLight extends Component {
  getColor = () => {
    const { status, } = this.props;

    if (status === "OK") return classes.ok;
    if (status === "FAILED") return classes.failed;
    if (status === "PARTIALLY_FAILED") return classes.partiallyFailed;
  };

  render() {
    const { status, } = this.props;
    return (
      <Root id={"basisTestTooltip"}>
        <Tooltip title={status} placement="right" >
          {status === "RUNNING" ? (
            <CircularProgress
              className={classes.progress}
              size={30}
              color="secondary"
            />
          ) : (
            <span className={this.getColor()} />
          )}
        </Tooltip>
      </Root>
    );
  }
}

TrafficLight.propTypes = {};

export default (TrafficLight);
