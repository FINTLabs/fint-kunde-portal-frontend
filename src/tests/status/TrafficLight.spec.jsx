import React from "react";
import { create } from "react-test-renderer";
import TrafficLight from "../../common/status/TrafficLight";

describe("TrafficLight with status FAILED", () => {
  it("has className containing failed", () => {
    const component = create(<TrafficLight status="FAILED" />);
    expect(component.root.findByType("span").props.className).toContain(
      "-failed-"
    );
  });
});

describe("TrafficLight with status OK", () => {
  it("has className containing ok", () => {
    const component = create(<TrafficLight status="OK" />);
    expect(component.root.findByType("span").props.className).toContain("-ok-");
  });
});

describe("TrafficLight with status PARTIALLY_FAILED", () => {
  it("has className containing partially_failed", () => {
    const component = create(<TrafficLight status="PARTIALLY_FAILED" />);
    expect(component.root.findByType("span").props.className).toContain(
      "-partiallyFailed-"
    );
  });
});

describe("TrafficLight with status RUNNING", () => {
  it("is of element type circle", () => {
    const component = create(<TrafficLight status="RUNNING" />);
    expect(component.root.findByType("circle")).toBeDefined();
  });
});

describe("Component", () => {
  const component = create(<TrafficLight status="RUNNING" />);
  it("should match snapshot", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });

});
