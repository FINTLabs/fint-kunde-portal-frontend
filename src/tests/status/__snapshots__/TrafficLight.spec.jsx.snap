// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should match snapshot 1`] = `
<div>
  <div
    aria-describedby={null}
    className="MuiCircularProgress-root TrafficLight-progress-4 MuiCircularProgress-colorSecondary MuiCircularProgress-indeterminate"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseLeave={[Function]}
    onMouseOver={[Function]}
    onTouchEnd={[Function]}
    onTouchStart={[Function]}
    role="progressbar"
    style={
      Object {
        "height": 30,
        "width": 30,
      }
    }
    title="RUNNING"
  >
    <svg
      className="MuiCircularProgress-svg"
      viewBox="22 22 44 44"
    >
      <circle
        className="MuiCircularProgress-circle MuiCircularProgress-circleIndeterminate"
        cx={44}
        cy={44}
        fill="none"
        r={20.2}
        strokeWidth={3.6}
        style={Object {}}
      />
    </svg>
  </div>
</div>
`;
